src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationExtensions.cs

```diff
@@ -44,7 +44,8 @@ internal static PEMethodSymbol GetSourceMethod(this CSharpCompilation compilatio
                         methodHandle = candidateMethod.Handle;
                         string stateMachineTypeName;
                         if (module.HasStringValuedAttribute(methodHandle, AttributeDescription.AsyncStateMachineAttribute, out stateMachineTypeName) ||
-                            module.HasStringValuedAttribute(methodHandle, AttributeDescription.IteratorStateMachineAttribute, out stateMachineTypeName))
+                            module.HasStringValuedAttribute(methodHandle, AttributeDescription.IteratorStateMachineAttribute, out stateMachineTypeName) ||
+                            module.HasStringValuedAttribute(methodHandle, AttributeDescription.AsyncIteratorStateMachineAttribute, out stateMachineTypeName))
```

Comment:
Thanks Chuck.
Do you know if IAsynEnumerable is supported in VB? I'm not sure how to get the equivalent VB test working.

Reply:
I'm not sure if iterator methods are supported with `IAsyncEnumerable` in VB. And it looks like `AsyncStreamsTypes` is defined in the `CSharpTestBase` base class, in C#, so it would require more effort to add a test even if supported. cc @jcouv.

Reply:
Async streams aren't supported in VB